libs/src/api/rewards.js

```diff
@@ -454,11 +452,15 @@ class Rewards extends Base {
 
     try {
       const response = await axios(request)
+      // if attestation is successful, 'result' represents a signature
+      // otherwise, 'result' is false
+      // - there may or may not be a value for `needs` if the attestation fails
+      // - depending on whether the user can take an action to attempt remediation
       const { result, needs } = response.data
 
-      if (needs) {
+      if (!result) {
         logger.error(`Failed to get AAO attestation: needs ${needs}`)
-        const mappedErr = GetAttestationError[needs] || GetAttestationError.AAO_ATTESTATION_UNKNOWN_RESPONSE
+        const mappedErr = GetAttestationError[needs] || GetAttestationError.AAO_ATTESTATION_FAILURE
```

Comment:
Seems like maybe there are 3 cases:
- Indexes into GetAttestationError when needs exists
- needs exists but not recognized - AAO_ATTESTATION_UNKNOWN_RESPONSE
- no needs: some general rejection error?